### PR TITLE
nfd: update to v0.5.0

### PR DESCRIPTION
**Pull request by przemeklal**
* update Helm chart to support NFD's master-worker model
* allow custom config to be passed using Helm values
* add support for TLS: disabled, autogenerated self-signed cert/keys (default) and user-provided cert/keys/CA
* update Ansible tasks and templates to use the updated Helm chart
* change default namespace from kube-system to node-feature-discovery

Signed-off-by: Przemyslaw Lal <przemyslawx.lal@intel.com>

### CODE REVIEW

**Comment by przemeklal** [comment]
Hey @mrymsza, @michaelspedersen, could you please take a look?

File: roles/nfd-install/charts/node-feature-discovery/templates/service.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  name: {{ template "node-feature-discovery.fullname" . }}-master
+  labels:
+{{ include "node-feature-discovery.labels" . | indent 4 }}
+spec:
+  selector:
+    app: {{ template "node-feature-discovery.fullname" . }}-master
+  ports:
+  - protocol: TCP
+    port: 8080
```
Comment (mrymsza) [inline roles/nfd-install/charts/node-feature-discovery/templates/service.yml]: Should the port number be hardcoded?

**Comment by michaelspedersen** [comment]
After looking through the changes a few times, generally lgtm.

File: roles/nfd-install/charts/node-feature-discovery/templates/service.yml (hunk 1 above)
Comment (przemeklal) [reply roles/nfd-install/charts/node-feature-discovery/templates/service.yml]: done

**Pull request merged**